sdk/python/lib/pulumi/runtime/rpc.py

```diff
@@ -571,10 +571,12 @@ def deserialize_properties(
 def deserialize_resource(
     ref_struct: struct_pb2.Struct, keep_unknowns: Optional[bool] = None
 ) -> "Resource":
-    urn = ref_struct["urn"]
-    version = ref_struct["packageVersion"] if "packageVersion" in ref_struct else ""
+    urn = str(ref_struct["urn"])
+    version = (
+        str(ref_struct["packageVersion"]) if "packageVersion" in ref_struct else ""
+    )
 
-    urn_parts = urn_util._parse_urn(urn)
+    urn_parts = urn_util._parse_urn(str(urn))
```

Comment:
This looks redundant with `urn = str(...)`.

```suggestion
    urn_parts = urn_util._parse_urn(urn)
```